consensus/src/main/java/org/apache/iotdb/consensus/statemachine/IStateMachine.java

```diff
@@ -34,7 +37,50 @@ interface Registry extends Function<ConsensusGroupId, IStateMachine> {}
 
   void stop();
 
+  /**
+   * apply a write-request from user
+   *
+   * @param IConsensusRequest write request
+   */
   TSStatus write(IConsensusRequest IConsensusRequest);
 
+  /**
+   * read local data and return
+   *
+   * @param IConsensusRequest read request
+   */
   DataSet read(IConsensusRequest IConsensusRequest);
+
+  /**
+   * IConsensus will periodically take the snapshot on both log and statemachine Data
+   *
+   * @param metadata the metadata IConsensus want IStateMachine to preserve
```

Comment:
Javadoc states that the updated snapshot should have lexicographically larger metadata, which needs to be supported by various consensus algorithm implementations

Reply:
done